### ISSUE TITLE
Catboost Compare function in combination with a for loop and Multiclass classification?

### ISSUE THREAD

**Issue by JPandela** [catboost/catboost#1454]
Problem: 

I'm pretty new to Catboost, so this might be a problem or just something that I am doing wrong.
I tried using the [compare](https://catboost.ai/docs/concepts/python-reference_catboostclassifier_modelcompare.html) method .
First of all, the given example seems to be outdated: 

![image](https://user-images.githubusercontent.com/72863447/95983246-97e4dc80-0e21-11eb-975b-46b8d9a45db4.png)

This can be fixed by just adding a directory to the models:

![image](https://user-images.githubusercontent.com/72863447/95983452-eabe9400-0e21-11eb-886b-4df7868e7076.png)

I managed to use this to compare 2 models, but my issue is that sometimes the compare only shows model 2  instead of both.
In the code below I train 2 models, the only difference being the train and validation set. 

`for fase in ['0','1','2','3']:`

        `###### first model, 

        params = {'iterations' : 1000,
                  'loss_function':'MultiClass', # objective function
                  'eval_metric':'MultiClass', # metric
                  'train_dir':'Models/Model1',
                  'use_best_model' : False,
                  'nan_mode': 'Max',
                  'cat_features' : cat_features,
                  'ignored_features': ign_features,
                  'class_weights':class_weights, # to cope with the imbalance between classes
                  'verbose': 200, # output to stdout info about training process every 200 iterations
                  'random_seed': SEED,
                 }
        cbc_pred = CatBoostClassifier(**params)

        cbc_pred.fit(X_train, y_train, eval_set=(X_valid, y_valid), plot=True  );

        ###### second model
        class_weights = class_weight.compute_class_weight(class_weight='balanced', classes=np.unique(y), y=y)

        params2 = {'iterations' : 1000,
                  'loss_function':'MultiClass', # objective function
                  'eval_metric':'MultiClass', # metric
                  'train_dir':'Models/Model2',
                  'use_best_model' : False,
                  'nan_mode': 'Max',
                  'cat_features' : cat_features,
                  'ignored_features': ign_features,
                  'class_weights':class_weights, # to cope with the imbalance between classes
                  'verbose': 200, # output to stdout info about training process every 200 iterations
                  'random_seed': SEED,
                 }
        cbc_pred2 = CatBoostClassifier(**params2)

        cbc_pred2.fit(X_train2, y_train2, eval_set=(X_valid2, y_valid2), plot=True );
        
        cbc_pred.compare(cbc_pred2, data = eval_dataset, metrics = ['MultiClass','MultiClassOneVsAll', 'Accuracy'],ntree_start=0, ntree_end = 1000, eval_period = 5)`

As you can see I loop over 4 Fases. 2 of the 4 times the compare is executed it shows both models and in the other cases it only shows model2.

output1:
![image](https://user-images.githubusercontent.com/72863447/95985677-26a72880-0e25-11eb-888a-4e2ddef00546.png)
output2:
![image](https://user-images.githubusercontent.com/72863447/95985507-f790b700-0e24-11eb-8a7c-6105e74da99c.png)
output3:
![image](https://user-images.githubusercontent.com/72863447/95985551-037c7900-0e25-11eb-8c4f-cdc9a54e7707.png)
output4:
![image](https://user-images.githubusercontent.com/72863447/95985580-0d05e100-0e25-11eb-91a4-7635716f1d05.png)

Additionally, could someone help explain me the fact that even if my first model trained better than my second, the second model still outperforms on the test set. The next 3 images show training of the first model, training of the second model and the compare of both.

![image](https://user-images.githubusercontent.com/72863447/95986010-a208da00-0e25-11eb-9c03-d0106dcfdcb9.png)
![image](https://user-images.githubusercontent.com/72863447/95986039-ac2ad880-0e25-11eb-8b97-ffb491362b1d.png)
![image](https://user-images.githubusercontent.com/72863447/95986099-c1076c00-0e25-11eb-802e-9a72926eeb70.png)

I know its a handfull to read, I would be gratefull for any tips.

catboost version: 0.14.2
Operating System: Windows 10

**Comment by ljluestc** [comment]
```
import catboost as cb

# ... your code ...

cbc_pred.fit(X_train, y_train, eval_set=(X_valid, y_valid), plot=True)

cbc_pred2.fit(X_train2, y_train2, eval_set=(X_valid2, y_valid2), plot=True)

cbc_pred.compare(cbc_pred2, data=eval_dataset, metrics=['MultiClass', 'MultiClassOneVsAll', 'Accuracy'], 
                 ntree_start=0, ntree_end=1000, eval_period=5, plot=True)

```
